src/App.jsx

```diff
@@ -1,247 +1,257 @@
-                2
-              </td>
+                  </th>
 
-              <td data-cy="ProductName">Bread</td>
-              <td data-cy="ProductCategory">🍞 - Grocery</td>
+                  <th>
+                    <span className="is-flex is-flex-wrap-nowrap">
+                      Category
 
-              <td
-                data-cy="ProductUser"
-                className="has-text-danger"
-              >
-                Anna
-              </td>
-            </tr>
-
-            <tr data-cy="Product">
-              <td className="has-text-weight-bold" data-cy="ProductId">
-                3
-              </td>
+                      <a href="#/">
+                        <span className="icon">
+                          <i data-cy="SortIcon" className="fas fa-sort-up" />
+                        </span>
+                      </a>
+                    </span>
+                  </th>
 
-              <td data-cy="ProductName">iPhone</td>
-              <td data-cy="ProductCategory">💻 - Electronics</td>
+                  <th>
+                    <span className="is-flex is-flex-wrap-nowrap">
+                      User
 
-              <td
-                data-cy="ProductUser"
-                className="has-text-link"
-              >
-                Roma
-              </td>
-            </tr>
-          </tbody>
-        </table>
+                      <a href="#/">
+                        <span className="icon">
+                          <i data-cy="SortIcon" className="fas fa-sort" />
+                        </span>
+                      </a>
+                    </span>
+                  </th>
+                </tr>
+              </thead>
+            ) : null}
+            <tbody>
+              {visibleProducts.length > 0 ? (
+                visibleProducts.map(product => (
+                  <tr data-cy="Product">
+                    <td className="has-text-weight-bold" data-cy="ProductId">
+                      {product.productId}
+                    </td>
+
+                    <td data-cy="ProductName">
+                      {product.productName}
+                    </td>
+
+                    <td data-cy="ProductCategory">
+                      {product.categoryIcon}
+                      <span> - </span>
+                      {product.categoryName}
+                    </td>
+
+                    <td
+                      data-cy="ProductUser"
+                      className={product.userSex === 'm'
+                        ? 'has-text-link'
+                        : 'has-text-danger'}
+                    >
+                      {product.userName}
+                    </td>
+                  </tr>
+                ))
+              ) : (
+                <tr>
+                  <td colSpan="4">
+                    <p data-cy="NoMatchingMessage">
+                      No products matching selected criteria
+                    </p>
+                  </td>
+                </tr>
```

Comment:
It should be `tr`. If there are no products... you shouldn't render the table, just `<p>...</p>`